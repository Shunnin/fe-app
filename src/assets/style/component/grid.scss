/*======================================

  Grid Component
    -> Descriptions

======================================*/

/*--------------------------------------
  Container Widths
--------------------------------------*/

// Set the container width, and override it for fixed navbars in media queries.
.container {
  @include container-fixed;

  @media (min-width: $screen-sm-min) {
    max-width: $container-sm;
  }

  @media (min-width: $screen-md-min) {
    max-width: $container-md;
  }

  @media (min-width: $screen-lg-min) {
    max-width: $container-lg;
  }
}

/*--------------------------------------
  Fluid Container
--------------------------------------*/

// Utilizes the mixin meant for fixed width containers, but without any defined
// width for fluid, full width layouts.
.container-fluid {
  @include container-fixed;
}
