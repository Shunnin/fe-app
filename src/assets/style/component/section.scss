$section-padding: $b-space-2x;

/* ----- Section Header ----- */
.section {
  padding: $b-space;

  /* ----- Section Title ----- */
  &__title {
    margin: $section-padding;
    font-weight: $b-font-weight-medium;
    font-size: $b-line-height-heading;
  }

  /* ----- Section Content ----- */
  &__content {
    @include clearfix();

    padding: $b-space-2x;
  }
}
