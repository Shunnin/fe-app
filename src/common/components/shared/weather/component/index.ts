export { Temperature } from './temperature/temperature.component';
export { WeatherCard } from './weather-card/weather-card.component';
export { WeatherItem } from './weather-item/weather-item.component';
export * from './weather-icon';
