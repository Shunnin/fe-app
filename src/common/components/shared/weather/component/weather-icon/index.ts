export { WeatherIcon } from './weather-icon.component';
export * from './weather-icon.constant';
