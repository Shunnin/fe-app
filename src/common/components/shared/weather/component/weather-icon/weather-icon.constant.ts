import { ReactComponent as WCloudyIcon } from '../../../../../../assets/images/weather/cloudy.svg';
import { ReactComponent as WHazeIcon } from '../../../../../../assets/images/weather/haze.svg';
import { ReactComponent as WHeavyRainIcon } from '../../../../../../assets/images/weather/heavy-rain.svg';
import { ReactComponent as WPartlyCloudyIcon } from '../../../../../../assets/images/weather/partly-cloudy.svg';
import { ReactComponent as WRainIcon } from '../../../../../../assets/images/weather/rain.svg';
import { ReactComponent as WSleetIcon } from '../../../../../../assets/images/weather/sleet.svg';
import { ReactComponent as WSnowIcon } from '../../../../../../assets/images/weather/snow.svg';
import { ReactComponent as WSunnyIcon } from '../../../../../../assets/images/weather/sunny.svg';
import { ReactComponent as WThunderstormIcon } from '../../../../../../assets/images/weather/thunderstorm.svg';
import { ReactComponent as WHighIcon } from '../../../../../../assets/images/weather/high-icon.svg';
import { ReactComponent as WHumidityIcon } from '../../../../../../assets/images/weather/humidity-icon.svg';
import { ReactComponent as WLowIcon } from '../../../../../../assets/images/weather/low-icon.svg';
import { ReactComponent as WPressureIcon } from '../../../../../../assets/images/weather/pressure-icon.svg';
import { ReactComponent as WWindIcon } from '../../../../../../assets/images/weather/wind-icon.svg';

export const CloudyIcon = WCloudyIcon;
export const HazeIcon = WHazeIcon;
export const HeavyRainIcon = WHeavyRainIcon;
export const PartlyCloudyIcon = WPartlyCloudyIcon;
export const RainIcon = WRainIcon;
export const SleetIcon = WSleetIcon;
export const SnowIcon = WSnowIcon;
export const SunnyIcon = WSunnyIcon;
export const ThunderstormIcon = WThunderstormIcon;
export const HighIcon = WHighIcon;
export const HumidityIcon = WHumidityIcon;
export const LowIcon = WLowIcon;
export const PressureIcon = WPressureIcon;
export const WindIcon = WWindIcon;
