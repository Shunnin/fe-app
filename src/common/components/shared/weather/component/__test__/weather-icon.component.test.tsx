import { render } from '@testing-library/react';

import { WeatherIcon } from '../weather-icon/weather-icon.component';

describe('<WeatherIcon />', () => {
  it.each([
    [800],
    [801],
    [802],
    [803],
    [804],
    [500],
    [501],
    [520],
    [521],
    [511],
    [502],
    [503],
    [504],
    [522],
    [531],
    [300],
    [301],
    [302],
    [310],
    [311],
    [312],
    [313],
    [314],
    [321],
    [200],
    [201],
    [202],
    [210],
    [211],
    [212],
    [221],
    [230],
    [231],
    [232],
    [600],
    [601],
    [602],
    [612],
    [613],
    [615],
    [616],
    [620],
    [621],
    [622],
    [611],
    [701],
    [711],
    [721],
    [731],
    [741],
    [751],
    [761],
    [762],
    [771],
    [781],
    [99999],
  ])('renders weather icon and matches the snapshot when given weather code is %p', code => {
    const { container } = render(<WeatherIcon code={code} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders big weather icon and matches the snapshot', () => {
    const { container } = render(<WeatherIcon code={800} big />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
