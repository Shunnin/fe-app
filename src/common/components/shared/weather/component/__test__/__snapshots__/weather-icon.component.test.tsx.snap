// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherIcon /> renders big weather icon and matches the snapshot 1`] = `
<reactcomponent
  style="width: 100px; height: 100px;"
/>
`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 200 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 201 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 202 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 210 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 211 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 212 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 221 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 230 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 231 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 232 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 300 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 301 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 302 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 310 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 311 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 312 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 313 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 314 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 321 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 500 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 501 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 502 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 503 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 504 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 511 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 520 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 521 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 522 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 531 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 600 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 601 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 602 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 611 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 612 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 613 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 615 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 616 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 620 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 621 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 622 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 701 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 711 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 721 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 731 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 741 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 751 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 761 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 762 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 771 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 781 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 800 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 801 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 802 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 803 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 804 1`] = `<reactcomponent />`;

exports[`<WeatherIcon /> renders weather icon and matches the snapshot when given weather code is 99999 1`] = `<reactcomponent />`;
